src/qibocal/protocols/characterization/ramsey.py

```diff
@@ -155,15 +156,19 @@ def _acquisition(
             ExecutionParameters(
                 nshots=params.nshots,
                 relaxation_time=params.relaxation_time,
-                acquisition_type=AcquisitionType.INTEGRATION,
-                averaging_mode=AveragingMode.CYCLIC,
+                acquisition_type=AcquisitionType.DISCRIMINATION,
+                averaging_mode=AveragingMode.SINGLESHOT,
             ),
             sweeper,
         )
         for qubit in qubits:
-            result = results[ro_pulses[qubit].serial]
+            probs = results[qubit].probability()
+            errors = [np.sqrt(prob * (1 - prob) / params.nshots) for prob in probs]
             data.register_qubit(
-                qubit, wait=waits, msr=result.magnitude, phase=result.phase
+                qubit,
+                wait=np.array(waits),
+                prob=np.array(probs),
+                errors=np.array(errors),
```

Comment:
```suggestion
                wait=waits,
                prob=probs,
                errors=errors,
```
It seems to work also without an explicit cast to `np.array`.